nilearn/image/resampling.py

```diff
@@ -455,8 +455,13 @@ def resample_img(img, target_affine=None, target_shape=None,
     target_shape = tuple(target_shape)
 
     if interpolation == 'continuous' and data.dtype.kind == 'i':
-        resampled_data_dtype = np.dtype(
-            data.dtype.name.replace('int', 'float'))
+        # cast unsupported data types to closest support dtype
+        aux = data.dtype.name.replace('int', 'float')
+        aux = aux.replace("ufloat", "float").replace("floatc", "float")
```

Comment:
> The rule is even simpler : int64 and float64 lead to float64. Anything
> else is float32.

I find that this is a bit fragile. I would rather test on something like

a.dtype.itemsize >= 8